libs/pxweb2-ui/src/lib/components/Table/Table.tsx

```diff
@@ -0,0 +1,301 @@
+import { PxTable } from '../../shared-types/pxTable';
+import { calculateRowAndColumnMeta, columnRowMeta } from './columnRowMeta';
+import { getPxTableData } from './cubeHelper';
+
+export interface TableProps {
+  pxtable: PxTable;
+}
+
+/**
+ * Represents the metadata for one dimension of a data cell.
+ */
+type DataCellMeta = {
+  varId: string; // id of variable
+  valCode: string; // value code
+  varPos: number; // variable position in stored data
+};
+
+/**
+ * Represents the metadata for multiple dimensions of a data cell.
+ */
+type DataCellCodes = DataCellMeta[];
+
+export function Table({ pxtable }: TableProps) {
+  const tableMeta: columnRowMeta = calculateRowAndColumnMeta(pxtable);
+  //console.log({ tableMeta });
+
+  const tableColumnSize: number = tableMeta.columns - tableMeta.columnOffset;
+  const headingDataCellCodes = new Array<DataCellCodes>(tableColumnSize); // Contains header variable and value codes for each column in the table
+  
+  // Create empty metadata structure for the dimensions in the header. 
+  // This structure will be filled with metadata when the header is created.
+  for (let i = 0; i < tableColumnSize; i++) {
```

Comment:
We think that using i is kind of standard for accessing data in an array. We have added comments for what i represents. In some of the loops we have kept the old iterator-names. This is because we are still developing this code and it will help us when we are looking at the old code. In the final version of the Table component we can look over all the for-loops and refactor it to be consistent with our code standard.